### PR TITLE
chore(market-matcher): market matcher next

### PR DESCRIPTION
**Pull request by kosecki123**
A re-designed version of the `market-matcher`

- no magic json configuration files
- no scheme generation - just JSON validation
- code refactoring
- stale simulator removal
- code strategies support
- uses dotenv configs

### CODE REVIEW

File: packages/market-matcher/scripts/buildSchemas.js [hunk 1]
```diff
@@ -76,8 +76,6 @@ async function run() {
 
     await generateSchemaIfNew('yarn --silent generate-rule-schema', '/schemas/rule.schema.json');
 
-    await generateSchemaIfNew('yarn --silent generate-conf-schema', '/schemas/conf.schema.json');
```
Comment (e00dan) [inline packages/market-matcher/scripts/buildSchemas.js:79]: May God bless for each schema less

File: packages/market-matcher/src/EntityStore.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        this.logger.verbose('Registered new supply #' + supply.id);
+    }
+
+    private registerAgreement(agreement: Agreement.Entity) {
+        const allowed = agreement.allowedMatcher.some(
+            matcherAddress => matcherAddress && matcherAddress.toLowerCase() === this.matcherAddress
+        );
+
+        if (!allowed) {
+            this.logger.verbose('This instance is not an matcher for agreement #' + agreement.id);
```
Comment (e00dan) [inline packages/market-matcher/src/EntityStore.ts:197]: `an matcher` => `a matcher`

**Review by e00dan — APPROVED** [review]
Impressive, great work!

File: packages/market-matcher/.env.example [hunk 3]
```diff
@@ -0,0 +1,6 @@
+WEB3 = "http://localhost:8545"
+MARKET_CONTRACT_ADDRESS = "0x35E1bA1dEC0bc5A14799563EF8564310846D1438"
+ORIGIN_CONTRACT_ADDRESS = "0xE377ae7241f76aC5Cdb89f5DEbc0dBE82f9c2C03"
+MATCHER_PUB_KEY = "0x585cc5c7829b1fd303ef5c019ed23815a205a59e"
```
Comment (josipbagaric) [inline packages/market-matcher/.env.example:4]: Do we need to have the public key defined? Can it not be derived from the private key?

File: packages/market-matcher/package.json [hunk 4]
```diff
@@ -14,20 +14,15 @@
     "version": "1.1.0",
     "main": "dist/js/src/index.js",
     "scripts": {
-        "build": "yarn generate-schemas && yarn build-ts",
+        "build": "yarn build-ts",
```
Comment (josipbagaric) [inline packages/market-matcher/package.json]: Can we just replace `build-ts` with `build`?

**Review by josipbagaric — APPROVED** [review]
Looks good. Would be cool to enforce types, as a lot of variables are untyped (copy-paste of old code, I assume).

File: packages/market-matcher/package.json (hunk 4 above)
Comment (kosecki123) [reply packages/market-matcher/package.json]: So build should clean and build-ts actually, which is here omitted.

File: packages/market-matcher/.env.example (hunk 3 above)
Comment (kosecki123) [reply packages/market-matcher/.env.example:4]: Yeah, good point.

**Comment by kosecki123** [comment]
> Looks good. Would be cool to enforce types, as a lot of variables are untyped (copy-paste of old code, I assume).

Thanks! `noImplicitAny` enabled for the `market-matcher` project now.

**Pull request merged**